src/app/home/user/services/user.service.ts

```diff
@@ -0,0 +1,33 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UserService {
+  constructor(private http: HttpClient) {}
+
+  updateUserDetails(data: Object) {
+    return this.http.put('http://localhost:8080/user/updateUserDetails', data, {
```

Comment:
`http://localhost:8080/user/....`
We will discuss how to remove the hardcode here.